File: app/src/main/java/org/breezyweather/daily/adapter/holder/PollenHolder.java
package org.breezyweather.daily.adapter.holder;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.res.ColorStateList;

import org.breezyweather.common.basic.models.options.unit.PollenUnit;
import org.breezyweather.common.basic.models.weather.Pollen;
import org.breezyweather.R;
import org.breezyweather.daily.adapter.DailyWeatherAdapter;
import org.breezyweather.daily.adapter.model.DailyPollen;
import org.breezyweather.databinding.ItemWeatherDailyPollenBinding;

public class PollenHolder extends DailyWeatherAdapter.ViewHolder {

    private final ItemWeatherDailyPollenBinding mBinding;
    private final PollenUnit mPollenUnit;

    public PollenHolder(ItemWeatherDailyPollenBinding binding) {
        super(binding.getRoot());
        mBinding = binding;
        mPollenUnit = PollenUnit.PPCM;
    }

    @SuppressLint({"SetTextI18n", "RestrictedApi"})
    @Override
    public void onBindView(DailyWeatherAdapter.ViewModel model, int position) {
        Context context = itemView.getContext();
        Pollen pollen = ((DailyPollen) model).getPollen();

        mBinding.grassIcon.setSupportImageTintList(ColorStateList.valueOf(
                pollen.getGrassColor(itemView.getContext())
        ));
        mBinding.grassTitle.setText(context.getString(R.string.grass));
        mBinding.grassValue.setText(
                mPollenUnit.getValueText(
                        context, pollen.getGrassIndex() != null ? pollen.getGrassIndex() : 0
                ) + " - " + pollen.getGrassDescription()
        );

        mBinding.ragweedIcon.setSupportImageTintList(ColorStateList.valueOf(
                pollen.getRagweedColor(itemView.getContext())
        ));
        mBinding.ragweedTitle.setText(context.getString(R.string.ragweed));
        mBinding.ragweedValue.setText(
                mPollenUnit.getValueText(
                        context, pollen.getRagweedIndex() != null ? pollen.getRagweedIndex() : 0
                ) + " - " + pollen.getRagweedDescription()
        );

        mBinding.treeIcon.setSupportImageTintList(ColorStateList.valueOf(
                pollen.getTreeColor(itemView.getContext())
        ));
        mBinding.treeTitle.setText(context.getString(R.string.tree));
        mBinding.treeValue.setText(
                mPollenUnit.getValueText(
                        context, pollen.getTreeIndex() != null ? pollen.getTreeIndex() : 0
                ) + " - " + pollen.getTreeDescription()
        );

        mBinding.moldIcon.setSupportImageTintList(ColorStateList.valueOf(
                pollen.getMoldColor(itemView.getContext())
        ));
        mBinding.moldTitle.setText(context.getString(R.string.mold));
        mBinding.moldValue.setText(
                mPollenUnit.getValueText(
                        context, pollen.getMoldIndex() != null ? pollen.getMoldIndex() : 0
                ) + " - " + pollen.getMoldDescription()
        );
    }
}

File: app/src/main/java/org/breezyweather/daily/adapter/holder/PollenHolder.kt
package org.breezyweather.daily.adapter.holder

import android.annotation.SuppressLint
import android.content.res.ColorStateList
import org.breezyweather.R
import org.breezyweather.common.basic.models.options.unit.PollenUnit
import org.breezyweather.daily.adapter.DailyWeatherAdapter
import org.breezyweather.daily.adapter.model.DailyPollen
import org.breezyweather.databinding.ItemWeatherDailyPollenBinding

class PollenHolder(
    private val mBinding: ItemWeatherDailyPollenBinding
) : DailyWeatherAdapter.ViewHolder(
    mBinding.root
) {
    private val mPollenUnit: PollenUnit = PollenUnit.PPCM

    @SuppressLint("SetTextI18n", "RestrictedApi")
    override fun onBindView(model: DailyWeatherAdapter.ViewModel, position: Int) {
        val context = itemView.context
        val pollen = (model as DailyPollen).pollen
        mBinding.grassIcon.supportImageTintList = ColorStateList.valueOf(
            pollen.getGrassColor(itemView.context)
        )
        mBinding.grassTitle.text = context.getString(R.string.allergen_grass)
        mBinding.grassValue.text = mPollenUnit.getValueText(
            context, pollen.grassIndex ?: 0
        ) + " - " + pollen.grassDescription
        mBinding.ragweedIcon.supportImageTintList = ColorStateList.valueOf(
            pollen.getRagweedColor(itemView.context)
        )
        mBinding.ragweedTitle.text = context.getString(R.string.allergen_ragweed)
        mBinding.ragweedValue.text = mPollenUnit.getValueText(
            context, pollen.ragweedIndex ?: 0
        ) + " - " + pollen.ragweedDescription
        mBinding.treeIcon.supportImageTintList = ColorStateList.valueOf(
            pollen.getTreeColor(itemView.context)
        )
        mBinding.treeTitle.text = context.getString(R.string.allergen_tree)
        mBinding.treeValue.text = mPollenUnit.getValueText(
            context, pollen.treeIndex ?: 0
        ) + " - " + pollen.treeDescription
        mBinding.moldIcon.supportImageTintList = ColorStateList.valueOf(
            pollen.getMoldColor(itemView.context)
        )
        mBinding.moldTitle.text = context.getString(R.string.allergen_mold)
        mBinding.moldValue.text = mPollenUnit.getValueText(
            context, pollen.moldIndex ?: 0
        ) + " - " + pollen.moldDescription
    }
}